integration/fuse/src/main/java/alluxio/fuse/file/FuseFileOutStream.java

```diff
@@ -154,18 +165,43 @@ public synchronized void truncate(long size) {
       mOutStream = Optional.of(AlluxioFuseUtils.createFile(mFileSystem, mAuthPolicy, mURI, mMode));
       return;
     }
+    if (mOutStream.isPresent() && size > currentSize) {
+      // support create() -> write() -> truncate(to larger value) -> write()
+      // but do not support append write workload
+      // e.g. file exist -> open(W or RW) -> truncate(to a larger value)
+      mExtendedFileLen = size;
+      return;
+    }
     throw new UnsupportedOperationException(
-        String.format("Cannot truncate file %s to size %s", mURI, size));
+        String.format("Cannot truncate file %s from size %s to size %s", mURI, currentSize, size));
   }
 
   @Override
   public synchronized void close() {
     if (mOutStream.isPresent()) {
       try {
+        long bytesWritten = mOutStream.get().getBytesWritten();
+        if (bytesWritten < mExtendedFileLen) {
+          fillEmptyBytes(mOutStream.get(), mExtendedFileLen - bytesWritten);
+        }
         mOutStream.get().close();
       } catch (IOException e) {
-        throw new RuntimeException(e);
+        throw new RuntimeException(
+            String.format("Failed to close the output stream of %s", mURI), e);
       }
     }
   }
+
+  private static void fillEmptyBytes(FileOutStream fileOutStream, long size) throws IOException {
+    int bufferSize = size > Integer.MAX_VALUE
```

Comment:
It is better to limit the size to xMB, or this might impact on the client stability.

For example, if the user is truncating a file to 100G, the buffer size would be 2G, that is too large.